Cython/Compiler/ModuleNode.py

```diff
@@ -1055,6 +1057,34 @@ def generate_enum_definition(self, entry, code):
             # Not pre-declared.
             code.putln("typedef enum %s %s;" % (name, name))
 
+    def generate_scoped_enum_definition(self, entry, code):
+        code.mark_pos(entry.pos)
+        type = entry.type
+        name = entry.cname or entry.name or ""
+        header, footer = self.sue_header_footer(type, "enum class", name)
+        code.putln(header)
+        enum_values = entry.enum_values
+        if not enum_values:
+            error(entry.pos, "Empty enum definition not allowed outside a 'cdef extern from' block")
```

Comment:
This seems fairly late for such an error. Isn't this something that the parser could detect? Or at least the declaration analysis?

Reply:
I moved the this check to the parser. Please note however, that for C-style enums, it is still being done during code generation. Happy to change that too if the parser code looks good to you.